### PR TITLE
React views

### PR DESCRIPTION
**Pull request by ardsouza**

### CODE REVIEW

File: client/js/views/file.react.jsx [hunk 1]
```diff
@@ -12,12 +12,14 @@ var File = React.createClass({
       diff : ''
     };
   },
-  componentWillReceiveProps: function(nextProps) {
+  componentWillReceiveProps: function(nextProps) { //selecting a new path changes the state of visualize which sets the props of file
```
Comment (dxuehu) [inline client/js/views/file.react.jsx]: remove comment

File: client/js/views/file.react.jsx [hunk 2]
```diff
@@ -27,12 +29,14 @@ var File = React.createClass({
     }.bind(this));
   },
 
-  componentWillMount: function() {
+  componentWillMount: function() { //on component creation
```
Comment (dxuehu) [inline client/js/views/file.react.jsx]: remove comment

File: client/js/views/file.react.jsx [hunk 3]
```diff
@@ -12,12 +12,14 @@ var File = React.createClass({
...
   },
-  componentWillReceiveProps: function(nextProps) {
+  componentWillReceiveProps: function(nextProps) { //selecting a new path changes the state of visualize which sets the props of file
     var currFile = this.props.filePaths[nextProps.currentPath];
+    //url props used by the diffualizer but not mainpage visualize
     var nextUrl = this.props.urls.to || currFile.raw_url;
     var url = this.props.urls.from || currFile.last_url || nextUrl;
     $.get(url)
     .always(function(data) {
+      //http requests to rawgit.com will sometimes fail but return the expected data in the responseText
```
Comment (dxuehu) [inline client/js/views/file.react.jsx]: //returned JS files will .fail, .always will catch them in the responseText

**Comment by tchan247** [comment]
lgtm

**Comment by dxuehu** [comment]
does not look good to me.

**Pull request merged**